src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -197,25 +264,47 @@ int main(int argc, char **argv)
 
   std::cout << " You called : " <<std::endl
             << argv[0] << std::endl
-            << "--imageDirectory " << sImageDir << std::endl
+            << "--combinationDirectory " << sCombinationDirectory << std::endl
+            << "--imageDirectories " << sImageDirs << std::endl
             << "--sensorWidthDatabase " << sfileDatabase << std::endl
             << "--outputDirectory " << sOutputDir << std::endl
             << "--focal " << focal_pixels << std::endl
-            << "--intrinsics " << sKmatrix << std::endl
+            << "--intrinsics " << sKmatrixs << std::endl
             << "--camera_model " << i_User_camera_model << std::endl
             << "--group_camera_model " << b_Group_camera_model << std::endl;
 
   // Expected properties for each image
-  double width = -1, height = -1, focal = -1, ppx = -1,  ppy = -1;
+  std::vector<double> focals, ppxs, ppys;
+  std::vector<std::string> ImageDirs;
+  std::vector<std::pair<std::string, size_t>> SortedImages;
 
   const EINTRINSIC e_User_camera_model = EINTRINSIC(i_User_camera_model);
 
-  if ( !stlplus::folder_exists( sImageDir ) )
-  {
-    std::cerr << "\nThe input directory doesn't exist" << std::endl;
+  if (!checkImageDirsStringValidity(sImageDirs, ImageDirs, sCombinationDirectory)) {
     return EXIT_FAILURE;
   }
 
+  if (ImageDirs.size() != 1) {
+    if ( !stlplus::folder_exists( sCombinationDirectory ) ) {
+      if ( !stlplus::folder_create( sCombinationDirectory ) ) {
+        std::cerr << "\nCannot create " << sCombinationDirectory << " directory" << std::endl;
+        return EXIT_FAILURE;
+      }
+    } else {
+        if (!std::experimental::filesystem::is_empty(sCombinationDirectory)) {
+          std::experimental::filesystem::remove_all(sCombinationDirectory);
+          if ( !stlplus::folder_create( sCombinationDirectory ) ) {
+            std::cerr << "\nCannot create " << sCombinationDirectory << " directory" << std::endl;
+            return EXIT_FAILURE;
+          }
+        }
+    }
+  } else {
+    sCombinationDirectory = ImageDirs[0];
+  }
+
+  SortedImages = aggregateImageDirs(ImageDirs, sCombinationDirectory);
```

Comment:
Change to auto sortedImages = ...;
Don't declare a variable until you have to use it.